crates/pera-types/src/transaction.rs

```diff
@@ -1292,7 +1292,7 @@ impl TransactionKind {
     /// TODO: use an iterator over references here instead of a Vec to avoid allocations.
     pub fn input_objects(&self) -> UserInputResult<Vec<InputObjectKind>> {
         let input_objects = match &self {
-            Self::ChangeEpoch(_) | Self::LockNextCommittee(..) => {
+            Self::ChangeEpoch(_) | Self::LockNextCommittee(..) | Self::DWalletMPCOutput(_) => {
```

Comment:
This can be problematic. You include an unnecessary shared object in every mpc output TX, despite it is only needed in the network dkg's tx. Please add a todo with an issue to check this.